compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -827,10 +823,30 @@ impl<'a, 'tcx> MirVisitor<'tcx> for MirNeighborCollector<'a, 'tcx> {
 
         let tcx = self.tcx;
         match terminator.kind {
-            mir::TerminatorKind::Call { ref func, .. } => {
+            mir::TerminatorKind::Call { ref func, ref args, .. } => {
                 let callee_ty = func.ty(self.body, tcx);
                 let callee_ty = self.monomorphize(callee_ty);
-                visit_fn_use(self.tcx, callee_ty, true, source, &mut self.output);
+                if let ty::FnDef(def_id, _) = callee_ty.kind()
+                        && self.tcx.fn_sig(*def_id).skip_binder().abi == RustIntrinsic
+                            && self.tcx.item_name(*def_id) == sym::const_eval_select
+                    {
+                        // mark the third arg as this function's neighbor.
+                        let callee_ty = args[2].ty(self.body, tcx);
+                        let mut callee_ty = self.monomorphize(callee_ty);
+                        if let ty::Closure(_, substs) = callee_ty.kind() {
```

Comment:
why is this checking for closures when compiler/rustc_typeck/src/check/fn_ctxt/checks.rs checks for FnDef and rejects anything else?